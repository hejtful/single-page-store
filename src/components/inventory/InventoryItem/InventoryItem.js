import React from 'react';

import styles from './InventoryItem.module.css';

export const InventoryItem = ({
  id,
  title,
  price,
  image,
  description,
  onChange,
  onRemove,
}) => {
  function handleSubmit(event) {
    event.preventDefault();
  }

  const handleChange = (field) => (event) => {
    onChange(id, field, event.target.value);
  };

  function handleRemove() {
    onRemove(id);
  }

  return (
    <div className={styles.wrapper}>
      <form onSubmit={handleSubmit}>
        <div className={styles.formGroupInline}>
          <input
            type="text"
            required
            value={title}
            placeholder="Title"
            onChange={handleChange('title')}
            className={styles.formElement}
          />

          <input
            type="number"
            step="0.01"
            min="0.01"
            required
            value={price}
            placeholder="Price"
            onChange={handleChange('price')}
            className={styles.formElement}
          />
        </div>

        <input
          type="url"
          required
          value={image}
          placeholder="Image URL"
          onChange={handleChange('image')}
          className={styles.formElement}
        />

        <textarea
          required
          value={description}
          placeholder="Description"
          onChange={handleChange('description')}
          className={`${styles.formElement} ${styles.textarea}`}
        />

        <button type="button" onClick={handleRemove} className={styles.button}>
          Remove Product
        </button>
      </form>
    </div>
  );
};
