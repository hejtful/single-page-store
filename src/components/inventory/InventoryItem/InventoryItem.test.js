import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import { InventoryItem } from './InventoryItem';

const initialInventoryItem = {
  id: '1',
  title: 'Test product title',
  price: 0.25,
  image: 'https://test.image.jpg',
  description: 'Test product description quite long',
};

const methods = {
  onChange: jest.fn(),
  onRemove: jest.fn(),
};

it('renders with initial values set', () => {
  const props = initialInventoryItem;

  const { container } = render(<InventoryItem {...props} {...methods} />);

  expect(container).toMatchSnapshot();
});

it('should call onChange method when an input value is changed', () => {
  const props = initialInventoryItem;

  const { getAllByRole } = render(<InventoryItem {...props} {...methods} />);

  userEvent.type(getAllByRole('textbox')[0], ' new');

  expect(methods.onChange).toBeCalledTimes(4);
});

it('should call onRemove method when remove button is clicked', () => {
  const props = initialInventoryItem;

  const { getByRole } = render(<InventoryItem {...props} {...methods} />);

  userEvent.click(getByRole('button'));

  expect(methods.onRemove).toBeCalled();
});
