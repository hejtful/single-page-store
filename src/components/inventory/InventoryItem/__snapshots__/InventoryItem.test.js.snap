// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with initial values set 1`] = `
<div>
  <div
    class="wrapper"
  >
    <form>
      <div
        class="formGroupInline"
      >
        <input
          class="formElement"
          placeholder="Title"
          required=""
          type="text"
          value="Test product title"
        />
        <input
          class="formElement"
          min="0.01"
          placeholder="Price"
          required=""
          step="0.01"
          type="number"
          value="0.25"
        />
      </div>
      <input
        class="formElement"
        placeholder="Image URL"
        required=""
        type="url"
        value="https://test.image.jpg"
      />
      <textarea
        class="formElement textarea"
        placeholder="Description"
        required=""
      >
        Test product description quite long
      </textarea>
      <button
        class="button"
        type="button"
      >
        Remove Product
      </button>
    </form>
  </div>
</div>
`;
